Adds missing lavaland/engine exports, minor refactor

81Denton opened the pull request:
:cl: Denton
add: Added missing export rewards for various lavaland items (tendril/megafauna/ruins) and engine parts.
balance: Increased export values for emitters, PA parts, field generators and radiation collectors to match the rest of engineering exports. Reduced supermatter shard value by 1000 credits.
rscdel: Removed export rewards for red/blue warp cubes since they're blacklisted from the cargo shuttle.
/:cl:

I noticed that a lot of lavaland and engineering related items had no export datums. 
I added the following:

**Engineering**: Tesla coil, grounding rod, tesla/singulo generator, AM control units+shielding.
**Tendrils**: Narnar/inquisition hardsuits, rod of asclepius, prison cube, voodoo doll, inferno, memento mori.
**Ruins**: Honkmother staff, bananium shoes, envy's knife, cursed russian revolver, veal render
**Megafauna**: Staff of storms, dragon's blood, angelic vocal cords, anomalous crystal, spellblade, HEV suit+helmet, drake armour.
(Not like miners will ever donate megafauna loot but w/e)

Also, engineering export costs were inconsistent, with the SM shard paying out 90% of its original purchase price, but other parts less than 50%.
I adjusted them so that they're worth roughly 75% of the original purchase price.

81Denton committed:
Adds missing lavaland/engine exports

ExcessiveUseOfCobblestone commented:
Boss drops should really be a singular (per boss) type

ExcessiveUseOfCobblestone commented:
Actually just make a generic "boss_drops/major" and "boss_drops/minor" and stick everything in those, most of them look like they're 40K regardless of boss difficulty.

81Denton commented:
Yeah, boss drops are 40k. Only the HEV suit is special with 30k for the suit and 10k for the helmet.

81Denton commented:
Actually let me create a separate file for lavaland items

ExcessiveUseOfCobblestone commented:
That's fine for their own type then.

I'd do that with the engine gens as well since they're all 4k.

Is there something that changes when you put them in a general type vs. unique? Maybe when you get the feedback that you've sold them or something?

81Denton committed:
refactors lavaland loot exports

ExcessiveUseOfCobblestone commented on code/modules/cargo/exports/lavaland.dm:
```diff
@@ -0,0 +1,57 @@
+						
+/datum/export/lavaland/megafauna/bubblegum_hev_suit
+	cost = 30000
+	unit_name = "Hostile Environment Suit"
```
`a... Hey, What the HECK is this?` :^)

81Denton replied:
lmao

81Denton committed:
thanks cobby

81Denton committed:
fixes HECK names

81Denton committed:
adds prison cube artifact

81Denton committed:
removes broken H.E.C.K. export printout, adds 10k ash drake armour export

81Denton committed:
reduces drake/legion export value, adds ruin loot

81Denton commented:
I had another look at the actual usefulness/rarity of lavaland items and changed their export values to be more fitting. I also added ruin loot to exports:

Nerfed the following:
- All ash drake loot is now only worth 10k credits instead of 40k, as they're easy to kill.
- Legion's staff of storms is down to 20k credits from 40k, same reason as above.
- Ship in a bottle is down to 10k from 20k.

Added additional items:
- Ash drake armor for 10k credits. I'm also fine with removing this, since each drake kill lets you craft two sets.
- Ruin loot for 10k credits: Honkmother staff, bananium shoes, envy's knife, cursed russian revolver, veal render.

SpaceManiac reviewed (dismissed):
Looks good to me

AnturK commented on code/modules/cargo/exports/lavaland.dm:
```diff
@@ -0,0 +1,65 @@
+						/obj/item/blood_contract,
+						/obj/item/gun/magic/staff/spellblade)
+
+/datum/export/lavaland/megafauna/total_printout() //in the unlikely case a miner feels like selling megafauna loot
```
You'll need to add new arguments from #39749

81Denton committed:
adds new arguments from #39749
